src/mesh/mod.rs

```diff
@@ -173,6 +189,40 @@ impl<Sh: Shape + Clone> ShapeMesh<Sh> {
 
         Ok(())
     }
+
+    // gibt den Punkt am Rand der Form zurück in der Richtung, in die wir gucken
+    pub fn get_focus(&self, camera: &Camera) -> Point3<f32> {
+        const EPSILON: f32 = 0.0001;
+
+        let mut pos = camera.position;
+        loop {
+            let distance = self.shape.distance(pos).min;
+            pos = pos + camera.direction() * distance;
+            if distance < EPSILON {
+                break;
+            }
+        }
+        pos
+    }
+}
+
+impl<Sh: Shape> EventHandler for ShapeMesh<Sh> {
+
+    fn handle_event(&mut self, e: &Event) -> EventResponse {
+        match e {
+            &Event::MouseInput(ElementState::Released, MouseButton::Left) => {
+                self.split_next_time = true;
+                return EventResponse::Continue;
+            },
+            &Event::KeyboardInput(ElementState::Released, _, Some(VirtualKeyCode::G)) => {
+                self.show_debug = !self.show_debug;
+                return EventResponse::Continue;
+            }
+            _ => {}
+        }
+        EventResponse::NotHandled
```

Comment:
Instead of doing a bunch of early returns here, the `match` could be last expression. Put the `NotHandled` into the default-match-case and remove the `return` from the other lines. That should be cleaner.